### PR TITLE
"Send Crash Report User Confirmation" Dialog after restart (iOS)

### PR DESCRIPTION
**Pull request by alex-oreshko**
SetUserConfirmationHander method added to Crashes SDK.
Also Puppet shows confirmation dialog after crash and restart using this method.
Call Crashes.SetUserConfirmationHandler(UserConfirmationHandler) in Awake() method to set handler.
Crashes service starts not in AppCenterStarter.original, but later, in AppCenterBehaviour->StartAppCenter(), to give app an opportunity to set handler in Awake()

### CODE REVIEW

File: Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.mm [hunk 1]
```diff
@@ -49,17 +50,25 @@ void appcenter_unity_crashes_disable_mach_exception_handler()
   [MSCrashes disableMachExceptionHandler];
 }
 
-MSErrorReport* app_center_unity_crashes_last_session_crash_report()
+void appcenter_unity_crashes_set_user_confirmation_handler(void(* userConfirmationHandler)())
 {
-    return [MSCrashes lastSessionCrashReport];
+    [MSCrashes setUserConfirmationHandler:^BOOL(NSArray<MSErrorReport *> *_Nonnull errorReports){
+        userConfirmationHandler();
+        return false;
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.mm]: Why do you not expose return value?
See description for this [here](https://github.com/Microsoft/AppCenter-SDK-Apple/blob/943515eac2dcbce7d2f72b0701f2d99f2df57f22/AppCenterCrashes/AppCenterCrashes/MSCrashes.h#L9)

File: Assets/Puppet/PuppetAppCenter.cs [hunk 2]
```diff
@@ -6,11 +6,28 @@
...
+
+    static PuppetAppCenter instance;
+
+    private void Awake()
+    {
+        Crashes.SetUserConfirmationHandler(UserConfirmationHandler);
+        instance = this;
+    }
+
+    [MonoPInvokeCallback(typeof(Crashes.UserConfirmationHandler))]
```
Comment (MatkovIvan) [inline Assets/Puppet/PuppetAppCenter.cs]: Is this work on UWP?

@evgeny-pol friendly reminder about CI

File: Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.mm (hunk 1 above)
Comment (alex-oreshko) [reply Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.mm]: Fixed, thank you

File: Assets/Puppet/PuppetAppCenter.cs (hunk 2 above)
Comment (alex-oreshko) [reply Assets/Puppet/PuppetAppCenter.cs]: Not sure, a have a Mac, UWP is not supported

**Review by evgeny-pol — APPROVED** [review]

**Pull request merged**